server/util/index.ts

```diff
@@ -14,3 +17,53 @@ export function formatAddress(address: string): string {
   }
   return addressString.formattedAddress;
 }
+
+function validateToken(
+  req: Request,
+  res: Response,
+  callback: (payload?: JWTPayload) => void,
+) {
+  const { authorization } = req.headers;
+  if (authorization) {
+    const [bearer, token] = authorization.split(' ');
+    if (bearer === 'Bearer') {
+      jwt.verify(token, process.env.JWT_SECRET!, (err, payload) => {
+        if (err) {
+          res.send({ err });
+        } else {
+          callback(payload as JWTPayload);
+        }
+      });
+    }
+  } else {
+    res.send({ err: 'No token provided' });
+  }
+}
+
+function isUserValid(userType: UserType, authLevel: UserType) {
+  const priority: { [type in UserType]: number } = {
+    User: 0,
+    Rider: 1,
+    Driver: 1,
+    Dispatcher: 2,
+  };
+  return userType === authLevel || priority[authLevel] < priority[userType];
```

Comment:
This is just a style thing. But if you wanted to make this line shorter, you could just use `priority[authLevel] <= priority[userType]`

Reply:
ah i did that to make sure any endpoints for riders can't be accessed by drivers and vice versa, do you think that's necessary?

Reply:
Oh I see. I'd keep it. It might not be necessary but I think it's a good thing to be able to distinguish between rider and driver permissions.